test/jasmine/tests/select_test.js

```diff
@@ -848,7 +848,60 @@ describe('Test select box and lasso per trace:', function() {
                     ]);
                     assertRanges([[1.66, 3.59], [0.69, 2.17]]);
                 },
-                null, BOXEVENTS, 'bar select'
+                null, BOXEVENTS, 'histogram select'
+            );
+        })
+        .catch(fail)
+        .then(done);
+    });
+
+    it('should work for box traces', function(done) {
+        var assertPoints = makeAssertPoints(['curveNumber', 'y']);
+        var assertRanges = makeAssertRanges();
+        var assertLassoPoints = makeAssertLassoPoints();
+
+        var fig = Lib.extendDeep({}, require('@mocks/box_grouped'));
+        fig.data.forEach(function(trace) {
+            trace.boxpoints = 'all';
+        });
+        fig.layout.dragmode = 'lasso';
+        fig.layout.width = 600;
+        fig.layout.height = 500;
+        addInvisible(fig);
+
+        Plotly.plot(gd, fig)
+        .then(function() {
+            return _run(
+                [[200, 200], [400, 200], [400, 350], [200, 350], [200, 200]],
+                function() {
+                    assertPoints([
+                        [0, 0.2], [0, 0.3], [0, 0.5], [0, 0.7],
+                        [1, 0.2], [1, 0.5], [1, 0.7], [1, 0.7],
+                        [2, 0.3], [2, 0.6], [2, 0.6]
+                    ]);
+                    assertLassoPoints([
+                        ['day 1', 'day 2', 'day 2', 'day 1', 'day 1'],
```

Comment:
@alexcjohnson Is this the desired behavior, or did an extra `c2d` conversion get forgotten? I believe outputting the linearized continuous category equivalent would make sense here.

Reply:
... we might have to wait for v2 though.

Reply:
I guess `c` format would make sense for applications that want to perhaps overlay something on the selected points... but I'd think the `d` value is what the user would expect (and what they would generally want to use), wouldn't it?

Reply:
My argument for `c`: 

`d` is _lossy_ for categories. Converting `c` to `d` can easily be done by the user, the opposite cannot (unless we officially expose `ax.d2c` and friends).

Reply:
So do you think the `x`/`y` coordinate should be converted to `d` too [here](https://github.com/plotly/plotly.js/pull/2094/files#diff-5862cfe1d69fa8963ebd7ee313bf9699R41)?

Reply:
> `d` is _lossy_ for categories.

These values had really better correspond to the actual data, not offset or jittered values... It's not important for category axes, but for *non-lossy* axes it is, otherwise you'll be pointed to the wrong data value! So in the end I don't think that *this* conversion is lossy, even though in general you're right.

> Converting `c` to `d` can easily be done by the user, the opposite cannot (unless we officially expose `ax.d2c` and friends).

How do you convert `c` to `d` for categories without `ax.c2d`? It depends on all sorts of details of the graph. Granted enough people use those methods already that I think we have to consider them exposed...

> So do you think the `x`/`y` coordinate should be converted to `d` too [here](https://github.com/plotly/plotly.js/pull/2094/files#diff-5862cfe1d69fa8963ebd7ee313bf9699R41)?

Good catch - yeah, I guess so. Everything we report back out with events should be `d`, I would think. That's a pretty sweeping statement though, so it's worth discussing.

Reply:
> How do you convert c to d for categories without ax.c2d?

Yeah, I guess that's kinda _hard_ to in general. Put that in the 9am brain-cramp bin.

> Everything we report back out with events should be d

I think so too. We should at the very least report everything in the same coordinate space. Right now, lasso coords are in `d` but point coords in are `c` even for [scatter](https://github.com/plotly/plotly.js/blob/master/src/traces/scatter/select.js#L45-L46) - which is ridiculous. I would consider this a bug, do you agree?

Reply:
> I would consider this a bug, do you agree?

🐞 
I hope not too many people depend on the current behavior with date axes, but yeah, I think we should unify on `d` ASAP. I guess the lesson is make sure there are tests on non-numeric axes whenever we add new features like this... otherwise the `c/d` distinction goes unnoticed.

Reply:
done in https://github.com/plotly/plotly.js/pull/2094/commits/8901528534936bf3d7697db5db6f01f24f256e50